### PR TITLE
Fix issues found by SonarCloud

### PR DESCRIPTION
**Pull request by Beacze**
https://www.ibm.com/developerworks/library/j-jtp05236/

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/forms/IndexingForm.java [hunk 1]
```diff
@@ -84,6 +84,7 @@ public void run() {
             try {
                 sleep(pause);
             } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/IndexingForm.java]: Move this line after logging the error, so in any case the error get logged.

File: Kitodo/src/main/java/de/sub/goobi/forms/MassImportForm.java [hunk 2]
```diff
@@ -344,10 +341,10 @@ public void uploadFile() throws IOException {
             basename = basename.substring(1);
         }
         if (basename.contains("/")) {
-            basename = basename.substring(basename.lastIndexOf("/") + 1);
+            basename = basename.substring(basename.lastIndexOf('/') + 1);
         }
         if (basename.contains("\\")) {
-            basename = basename.substring(basename.lastIndexOf("\\") + 1);
+            basename = basename.substring(basename.lastIndexOf('\\') + 1);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/MassImportForm.java]: Are you sure about this change? `lastIndexOf` can be called with a string type and int type value and in this case the string value must be used as we are checking for double backslash characters. Or did I missed something?

Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/forms/MassImportForm.java]: It is escape character, so this change works correctly.

Comment (henning-gerhardt) [reply Kitodo/src/main/java/de/sub/goobi/forms/MassImportForm.java]: Thank you for this explanation - escaping characters confuse me sometimes.

**Pull request merged**